cmake-modules/AzureBuildTargetForCI.cmake

```diff
@@ -32,6 +32,21 @@ macro(create_per_service_target_build_for_sample  service target)
     if(CMAKE_GENERATOR MATCHES "Visual Studio.*")
         SET(binary "Release/${binary}")
     endif()
-    file(APPEND ${CMAKE_BINARY_DIR}/${service}-samples.txt "${CMAKE_CURRENT_BINARY_DIR}/${binary}\n")
-    
+
+    set(RUN_SAMPLE TRUE)
+    if(${ARGC} GREATER 2)
+      set(extra_args "${ARGN}")
+      foreach(arg IN LISTS extra_args)
+        if (${arg} MATCHES "DISABLE_RUN")
+          set(RUN_SAMPLE FALSE)
+          message("Disabling sample execution for ${service}/${binary}")
+        else()
+          message(FATAL_ERROR "Unknown extra argument: ${arg}")
+        endif()
+      endforeach()
+
+      if (RUN_SAMPLE)
+          file(APPEND ${CMAKE_BINARY_DIR}/${service}-samples.txt "${CMAKE_CURRENT_BINARY_DIR}/${binary}\n")
+      endif()
```

Comment:
Should this if (RUN_SAMPLE) check be outside the containing if-block?

That way, if a sample doesn't specify DISABLE_RUN as an extra param, the default is that it is executes.
Currently, the way this is written, the sample only runs if ARGC > 2. I am assuming by default ARGC == 2, when DISABLE_RUN isn't specified.

Reply:
Wow, good catch. That's a huge bug.